app/assets/javascripts/controllers/reconfigure/reconfigure_form_controller.js

```diff
@@ -271,6 +293,41 @@ ManageIQ.angular.app.controller('reconfigureFormController', ['$http', '$scope',
     vm.updateNetworkAdaptersAddRemove();
   };
 
+  vm.editExistingNetworkAdapter = function(thisNetworkAdapter) {
+    thisNetworkAdapter.vLan_edit_requested = null;
+    thisNetworkAdapter.add_remove = 'update';
+    thisNetworkAdapter.originalVLan = thisNetworkAdapter.vlan;
+    vm.updateNetworkAdaptersAddRemove();
+  };
+
+  vm.cancelChangeExistingNetworkAdapter = function(thisNetworkAdapter) {
+    thisNetworkAdapter.vLan_edit_requested = null;
+    thisNetworkAdapter.vlan = thisNetworkAdapter.originalVLan;
+    thisNetworkAdapter.add_remove = '';
+    vm.updateNetworkAdaptersAddRemove();
+  };
+
+  vm.confirmEditSelectedNetworkAdapter = function(networkAdapter) {
+    angular.forEach(vm.reconfigureModel.vmNetworkAdapters, function(nap) {
```

Comment:
If you want to mutate the input array you should use `map` to preserver immutability. You can use lodash `_.map`
https://lodash.com/docs/4.17.15#map

Reply:
I will take a look

Reply:
Found better solution not needing the loop